app/javascript/src/NewService/components/FormTemplate.jsx

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import React from 'react'
+import {CSRFToken} from 'utilities/utils'
+import type {ServiceDiscoveryFormTemplateProps, FormTemplateProps} from 'NewService/types'
+
+const getServiceDiscoveryProps = (): ServiceDiscoveryFormTemplateProps => {
+  return {
+    hiddenServiceDiscoveryInput: {
+      value: 'discover',
+      name: 'service[source]',
+      id: 'service_source'
+    },
+    serviceDiscoverySelect: {
+      namespace: {
+        htmlFor: 'namespace',
+        label: 'Namespace',
+        name: 'service[namespace]',
+        id: 'service_namespace'
+      },
+      projects: {
+        htmlFor: 'service_name',
+        label: 'Name',
+        name: 'service[name]',
+        id: 'service_name'
+      }
+    }
+  }
+}
+
+const renderLabel = (namespace: {htmlFor: string, label: string}, required: boolean = false) => <label htmlFor={namespace.htmlFor}>{namespace.label}
+  { required && <abbr title="required">*</abbr> }
+</label>
+
+const renderSelectOptions = (options: string[]) => options.map(option => <option key={option} value={option}>{option}</option>)
+
+const renderSelect = (namespace: {name: string, id: string}, props: FormTemplateProps) => <select required="required" name={namespace.name} id={namespace.id} onChange={props.handleChangeNamespaces}
+  data-selected-value={namespace.id === 'service_namespace' ? props.selectedNamespace : ''}>
+  {props.namespaces && renderSelectOptions(props.namespaces)}
+</select>
+
+const ErrorMessage = ({error}) => <p style={{color: 'red'}}>{ error.errorMsg && `Sorry, your request has failed with the error: ${error.errorMsg}`}</p>
```

Comment:
@thomasmaas @guicassolato
If fetch fail it could be for several reasons, I think we could show the error message coming from the response, for example: `Sorry, your request has failed with the error: Error: Not Found`
Another error could be `TypeError: Network request failed` or `Error: Internal Server Error`